### ISSUE TITLE
[Bug]: rolling-update CN behind proxy cause prepared stmt lost

### ISSUE THREAD

**Issue by aylei** [matrixorigin/matrixone#16357]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.2-dev

### Commit ID

 1f8559d74

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Run TPCC and rolling update CNs behind proxy, tpcc client report stmt not found:

```
2024-05-23 07:41:44 FATAL jTPCCTerminal:328 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20400, ErrorMessage : invalid state prepared statement '__mo_stmt_id_2' does not exist
2024-05-23 07:41:44 FATAL jTPCCTerminal:297 - [UNEXPECTED][DELIVERY_BG][EXECUTION] ErrorCode : 20400, ErrorMessage : invalid state prepared statement '__mo_stmt_id_25' does not exist
2024-05-23 07:41:44 FATAL jTPCCTerminal:328 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20400, ErrorMessage : invalid state prepared statement '__mo_stmt_id_2' does not exist
2024-05-23 07:41:44 FATAL jTPCCTerminal:328 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20400, ErrorMessage : invalid state prepared statement '__mo_stmt_id_2' does not exist
2024-05-23 07:41:44 FATAL jTPCCTerminal:214 - [UNEXPECTED][TT_PAYMENT][EXECUTION] ErrorCode : 20400, ErrorMessage : invalid state prepared statement '__mo_stmt_id_16' does not exist
2024-05-23 07:41:44 FATAL jTPCCTerminal:328 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20400, ErrorMessage : invalid state prepared statement '__mo_stmt_id_2' does not exist
2024-05-23 07:41:44 FATAL jTPCCTerminal:328 - [UNEXPECTED][TT_NEW_ORDER][EXECUTION] ErrorCode : 20400, ErrorMessage : invalid state prepared statement '__mo_stmt_id_2' does not exist
```

### Expected Behavior

Prepared stmt should be migrated when gracefully draining CN.

### Steps to Reproduce

```Markdown
1. Setup a cluster with proxy and multiple CNs
2. Run `tpcc` test
3. Rolling-update CNs
```

**Comment by volgariver6** [comment]
fixed

**Comment by aylei** [comment]
https://github.com/matrixone-cloud/mocloud-tester/actions/runs/9309219505/job/25624214139

dashboard: https://grafana.cn-dev.matrixone.tech/goto/e1Nv3NsIg?orgId=1

**Comment by volgariver6** [comment]
提交pr修改多次迁移的问题，但即使多次迁移也不应该发生错误，所以也加了日志，再观察一下

**Comment by volgariver6** [comment]
在proxy做迁移session的时候，重新选择的cn不能跟原来的cn一样：
1. 迁到原来的cn没有必要
2. 根据conn id建立的session，如果迁移到同一个cn上，会导致有冲突

所以 Route() 需要确保返回的cn跟之前的cn是不一样的。@aylei 麻烦处理一下这里的问题

**Comment by aylei** [comment]
closed by #16689
